homework6.py

```diff
@@ -0,0 +1,194 @@
+import json
+import os
+import pprint
+
+class Ipv4_manager:
+    """ предоставляет ф-ции для работы с ipv4 """
+    _sep = '.'
+
+    def __init__(self, ip_list):
+        if all(self.__class__.is_ipv4(ip) for ip in ip_list):
+            raise ValueError("не верный формат ip-адреса")
+        self._ip_list = ip_list
+
+    @property
+    def ip_list(self):
+        return self._ip_list
+
+    def get_ip_reversed(self):
+        """ возвращает перевёрнутые ip-адресов"""
+        return [self.__class__._sep.join(ip.split(self.__class__._sep)[::-1]) for ip in self.ip_list]
+
+    def get_without_last_actets(self):
+        """ возвращает последние 3 актета"""
+        return [ip.rsplit(self.__class__._sep, 1)[0] for ip in self.ip_list]
+
+    def get_lst_actets(self):
+        """ возвращает первый актет """
+        return [ip.rsplit(self.__class__._sep, 1)[1] for ip in self.ip_list]
+
+    @staticmethod
+    def is_ipv4(ip):
+        """ проверяет соотвецтвует ли ip нужному формату """
+
+        if not isinstance(ip, str):
+            raise TypeError("ip-адресс должен быть строкой")
+        actet_list = ip.split(Ipv4_manager._sep)
+        if len(actet_list) != 4:
+            return False
+        if not all(isinstance(actet, int) for actet in actet_list):
+            return False
+        parsed_actet_list = [int(actet) for actet in actet_list]
+        if not all(0 <= actet <= 255 for actet in parsed_actet_list):
+            return False
+        return True
+
+
+class JSON_manager:
+    @staticmethod
+    def dump(path, data):
+        with open(path, 'w') as f:
+            json.dump(data, f)
+    
+    @staticmethod
+    def load(path):
+        if not os.path.exists(path):
+            raise FileNotFoundError("файл не найден")
+        with open(path) as f:
+            return json.load(f)
+
```

Comment:
It would be nice to add one more method for JSON validation to catch JSONDecodeError